identity-auth-core/src/main/scala/com/gu/identity/auth/IdentityClient.scala

```diff
@@ -0,0 +1,75 @@
+package com.gu.identity.auth
+
+import cats.effect._
+import io.circe.Decoder
+import io.circe.generic.semiauto.deriveDecoder
+import org.http4s.{Header, Headers, Request, Response, Status, Uri}
+import org.http4s.circe._
+import org.http4s.client.blaze.BlazeClientBuilder
+import com.gu.identity.model.User
+
+import scala.concurrent.ExecutionContext
+
+private class IdentityClient(identityApiUri: Uri, serverAccessToken: String)(implicit ec: ExecutionContext)
+  extends CirceEntityDecoder {
+
+  import IdentityClient._
+
+  private implicit val cs: ContextShift[IO] = IO.contextShift(ec)
+
+  private def executeGetRequest[A : Decoder](headers: Headers, path: String): IO[A] =
+    BlazeClientBuilder[IO](ec).resource.use { client =>
+      val request = Request[IO](uri = identityApiUri / path, headers = headers)
+      client.fetch(request)(decodeResponse[A])
+    }
+
+  private def headersForCredentials(credentials: UserCredentials): Headers =
```

Comment:
Are we expecting IDAPI to understand 3 new headers?

Reply:
no, mock values; apologies, should have put a TODO comment next to them to change these